projects/vdk-core/tests/vdk/internal/builtin_plugins/run/test_data_job.py

```diff
@@ -23,7 +23,8 @@ def test_run_when_step_fails():
 
     result = data_job.run()
     assert result.is_failed()
-    assert isinstance(result.exception, IndentationError)
+    assert result.exception is None
```

Comment:
Why should result.exception be None, as result.get_exception() should be IndentationError? Is this not contradictory?

Reply:
Hmm. I need to change `get_exception` name perhaps.

If a step fails - exception is recorded only for the step in StepResult
If the job fails for other reason (for example initialize_job fails or something else outside of the step fails) then we set ExecutionResult.exception is populated only then.

get_exception aims to figure out what is the appropriate exception to raise to the user - if there's ExectuionResult level exception then it raises it otherwise it raises the first StepResult exception . If none is found it raises its own exception.

Reply:
> I need to change get_exception name perhaps.


Any ideas?

Reply:
It seems to me that you already renamed it to get_exception_to_raise, which I think is a good candidate.